apps/api/src/infrastructure/repositories/blockWebsocket.ts

```diff
@@ -12,27 +12,109 @@ export class BlockWebsocketRepository
   implements BlockRepository
 {
   private web3: Web3 | null = null;
+  private provider: WebSocketProvider | null = null;
+  private websocketUrl!: string;
+  private noDataTolerance!: number;
+  private lastBlockTimestamp: number = 0;
+  private connectionTimestamp: number = 0;
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   private subscription: any | null = null;
 
-  execute(websocketUrl: string): void {
-    this.web3 = new Web3(websocketUrl);
+  execute(websocketUrl: string, noDataTolerance: number = 30_000): void {
+    this.websocketUrl = websocketUrl;
+    this.noDataTolerance = noDataTolerance;
+    this.connect();
+  }
+
+  connect(): void {
+    console.log('Trying to connect..');
+    this.web3 = new Web3(this.getProvider());
+  }
+
+  keepAlive(): void {
+    const intervalId = setInterval(() => {
+      if (this.connectionTimestamp == 0) {
```

Comment:
I think it's usually preferred

```suggestion
      if (this.connectionTimestamp === 0) {
```

Not for this PR, but perhaps you may want to add [our eslint config](https://www.npmjs.com/package/eslint-config-bloq)